components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/par/OAuth2ParEndpoint.java

```diff
@@ -146,4 +158,37 @@ private boolean isRequestUriProvided(MultivaluedMap<String, String> params) {
 
         return params.containsKey(OAuthConstants.OAuth20Params.REQUEST_URI);
     }
+
+    private void checkClientAuthentication(HttpServletRequest request) throws ParCoreException {
+
+        OAuthClientAuthnContext oAuthClientAuthnContext = getClientAuthnContext(request);
+        if (oAuthClientAuthnContext.isAuthenticated()) {
+            return;
+        }
+        if (StringUtils.isNotBlank(oAuthClientAuthnContext.getErrorCode())) {
+            if (oAuthClientAuthnContext.getErrorCode().equals(OAuth2ErrorCodes.SERVER_ERROR)) {
```

Comment:
let's compare the constant to the variable. Helps to avoid NPEs
```suggestion
            if (OAuth2ErrorCodes.SERVER_ERROR.equals(oAuthClientAuthnContext.getErrorCode())) {
```

Reply:
this way, we don't need to have the StringUtils.isNotBlank check